reaktive/src/commonMain/kotlin/com/badoo/reaktive/subject/behavior/BehaviorObservable.kt

```diff
@@ -0,0 +1,8 @@
+package com.badoo.reaktive.subject.behavior
+
+import com.badoo.reaktive.observable.Observable
+
+interface BehaviorObservable<out T> : Observable<T> {
+
+    val value: T
```

Comment:
should we further segregate this interface and create something like value provider and compose BehaviourObservable from Observable and value provider?

Reply:
Makes sense! Thanks! @mk-jones, @cherry-perry what do you think?

Reply:
Not sure.
When we need value provider, we use just kotlin lamda most of the time.

Reply:
Fare point)

Reply:
A would still avoid extracting the `value provider` interface at this point, but we can do it at any time later.

Reply:
@mk-jones What is your opinion?

Reply:
I don't yet see a use-case for the value provider interface. I wouldn't extract it without having a use-case in mind.

Reply:
Thanks!